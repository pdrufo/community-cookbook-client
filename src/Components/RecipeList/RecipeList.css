h2 {
  color: rgb(26, 29, 80);
  font-family: 'Libre Baskerville', serif;
  margin-top: 80px;
}

@media only screen and (min-width: 768px){ 
  
  h2{
    margin-top: 125px;
  }
}