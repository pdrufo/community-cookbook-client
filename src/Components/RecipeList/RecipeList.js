import React from 'react'
import RecipeListItem from '../RecipeListItem/RecipeListItem'
import './RecipeList.css'
import ApiContext from '../../ApiContext'


export default class RecipeList extends React.Component {

  static contextType = ApiContext;

  render() {
    const recipes = this.context.recipes.map((recipe, idx) =>
    <RecipeListItem key={idx} data={recipe} />
    );
  return (
    <div>
    <section className="recipe-search">
      <form className="recipe-search-form">
       <div className="form-section">
          <label htmlFor="recipe-search">Find a recipe:</label>
          <input type="text" name="recipe-search" />
       {/* </div>
       <div className="form-section"> */}
         <button type="submit">Submit</button>
      </div>
      </form>
    </section>
    <section className='recipe-list'>
      {recipes}
    </section>
    </div>
  )
}
}