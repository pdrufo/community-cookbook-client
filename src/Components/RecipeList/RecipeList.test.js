import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom'
import RecipeList from './RecipeList';
import renderer from 'react-test-renderer'


describe('GET /recipes' , () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><RecipeList /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter><RecipeList /></BrowserRouter>)
      .toJSON()
    expect(tree).toMatchSnapshot();
  })
})