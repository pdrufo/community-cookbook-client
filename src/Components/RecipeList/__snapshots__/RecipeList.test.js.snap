// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /recipes renders the UI as expected 1`] = `
<div>
  <section
    className="recipe-search"
  >
    <form
      className="recipe-search-form"
    >
      <div
        className="form-section"
      >
        <label
          htmlFor="recipe-search"
        >
          Find a recipe:
        </label>
        <input
          name="recipe-search"
          type="text"
        />
        <button
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </section>
  <section
    className="recipe-list"
  />
</div>
`;
